index.js

```diff
@@ -133,25 +95,69 @@ function constraintDirective () {
     })
 }
 
-const constraintDirectiveTypeDefs = /* GraphQL */`
-  directive @constraint(
-    # String constraints
-    minLength: Int
-    maxLength: Int
-    startsWith: String
-    endsWith: String
-    contains: String
-    notContains: String
-    pattern: String
-    format: String
-
-    # Number constraints
-    min: Float
-    max: Float
-    exclusiveMin: Float
-    exclusiveMax: Float
-    multipleOf: Float
-    uniqueTypeName: String
-  ) on INPUT_FIELD_DEFINITION | FIELD_DEFINITION | ARGUMENT_DEFINITION`
-
-module.exports = { constraintDirective, constraintDirectiveTypeDefs }
+function validateQuery (schema, query, variables, operationName) {
+  // console.log('validateQuery')
+  const typeInfo = new TypeInfo(schema)
+
+  const errors = []
+  const context = new ValidationContext(
+    schema,
+    query,
+    typeInfo,
+    (error) => errors.push(error)
+  )
+  const visitor = new QueryValidationVisitor(context, {
+    variables,
+    operationName
+  })
+
+  visit(query, visitWithTypeInfo(typeInfo, visitor))
+
+  return errors
+}
+
+function createApolloQueryValidationPlugin ({ schema }) {
+  return {
+    async requestDidStart () {
+      return ({
+        async didResolveOperation ({ request, document }) {
+          const query = request.operationName
+            ? separateOperations(document)[request.operationName]
+            : document
+
+          const errors = validateQuery(
+            schema,
+            query,
+            request.variables,
+            request.operationName
+          )
+          if (errors.length > 0) {
+            throw errors.map(err => {
+              const { UserInputError } = require('apollo-server-errors')
```

Comment:
Is there a reason we're require'ing inside the map? `apollo-server-errors` needs adding as a dependency or as a peer dependency?

Reply:
I do not want to add it always, as it is necessary only if Apollo plugin is used. `dependency` is added always, and `peerDependency` startin in npm 7 :-( So I have it only in `devDependency` and solve problem in other envs where it is not present this way.